### PR TITLE
Fix dom select[All] and trigger

### PR DESCRIPTION
**Pull request by giuseppeg**
Addresses #368

### CODE REVIEW

File: lib/dom.js [hunk 1]
```diff
@@ -89,7 +95,8 @@ define(function() {
 
   // Elements that can be jumped to directly from a string
   var selectorShortcut = {
-    document: document.documentElement,
+    document: document,
+    window: window,
```
Comment (giuseppeg) [inline lib/dom.js]: We should remove these imho
and add `'html': document.documentElement`

File: lib/logger.js [hunk 2]
```diff
@@ -15,8 +15,15 @@ define(
       off: 'x '
     };
 
+    var eventNameAliases = {
```
Comment (giuseppeg) [inline lib/logger.js]: ouch this should be `elemNameAliases` :D

**Comment by phuu** [comment]
Thanks for this @giuseppeg. Could you provide a bit more info about what's broken? Would help me out a lot.

**Comment by giuseppeg** [comment]
@phuu https://github.com/flightjs/flight/issues/368 let me know if you need further info.

File: lib/dom.js (hunk 1 above)
Comment (phuu) [inline lib/dom.js]: Hmm. I was trying to match what jQuery does but it looks like I got it wrong. So you're right, we can get rid of them and add an `html` key.

File: test/spec/events_spec.js [hunk 3]
```diff
@@ -219,6 +219,11 @@ define(['lib/component', 'lib/registry', 'lib/dom'], function (defineComponent,
       instance2.on(document, 'click', spy);
       instance1.trigger('click');
       expect(spy).toHaveBeenCalled();
+
+      spy = jasmine.createSpy();
```
Comment (phuu) [inline test/spec/events_spec.js]: Could you set this up as a separate test, perhaps that has a list of elements and generates a test for  each of them? `[window, document, 'head', 'body'].forEach(...)`

**Comment by phuu** [comment]
Thanks @giuseppeg. So what's still broken — the `('document', { ... })` case? Definitely we should test for that if it's a valid use of the API.

**Comment by giuseppeg** [comment]
@phuu my understanding is that trigger is broken when you fire with an event object – [relevant code](https://github.com/flightjs/flight/blob/master/lib/base.js#L100-L115).

Consider this:
```javascript
this.trigger(document, { type: "click" });
```

Here's what happens step by step:

```javascript
var element, type, data, event;
var lastIndex = arguments.length - 1, lastArg = arguments[lastIndex];

// lastIndex == 1
// lastArg == { type: "click" }

if (typeof lastArg != 'string') { // evaluates to true
    lastIndex--;
    data = lastArg;
}

// lastIndex == 0

if (lastIndex == 1) { // evaluates to false
    element = dom.select(arguments[0]);
    event = arguments[1];
} else {
    element = this.node;
    event = arguments[0];
}

// element == this.node instead of `document`
// event == document instead of `{ type: "click" }`
```

**Comment by giuseppeg** [comment]
@phuu I was thinking that maybe I can split this PR in two and we can merge the `select[All]` fix for now and fix `tigger` later.

If you agree what do you think about my suggestion to remove `document` and `window` from the `selectorShortcut`s and add `html`? This mean that `dom.select('document')` for example won't work anymore... I don't see why you would use `dom.select` to select `document` or `window` :)

**Comment by phuu** [comment]
@giuseppeg Totally reasonable, yeah — and I'm onboard with removing those string shortcuts too.

**Pull request closed without merging**